dht/dhtcore/NodeStore.c

```diff
@@ -1001,6 +1001,8 @@ static struct Node_Link* discoverLinkC(struct NodeStore_pvt* store,
     }
 
     struct Node_Two* parent = closest->child;
+    uint64_t realPathParentChild = pathParentChild == 1 ?
+        closest->cannonicalLabel : pathParentChild;
```

Comment:
Hmm also you may need to do this:
```c
uint64_t oldPath = child->address.path;
child->address.path = parent->address.path;
RumorMill_addNode(store->renumberMill, &child->address);
child->address.path = oldPath;
```
because otherwise it will fire a packet into outer space trying to find the child at it's previous location.